core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -1920,6 +1963,8 @@ public function editAll()
 	 */
 	public function source()
 	{
+		$this->denyAccessUnlessGranted(ContaoCorePermissions::DCA_EDIT, new DataContainerSubject($this->strTable, $this->intId));
```

Comment:
I think we should move the permission check after `isValid` just below, because we don't want permission checks on invalid files.

Reply:
Adjusted in 9705ea5111e1b53942ecbcb63fee0d49519bd017.